moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -1063,30 +1068,30 @@ bool IKFastKinematicsPlugin::getPositionIK(const geometry_msgs::Pose& ik_pose, c
     ROS_DEBUG_STREAM_NAMED("ikfast", "No IK solution");
   }
 
-  // Among the solutions under limits, find the one that is closest to ik_seed_state
-  if (solutions_obey_limits.size() != 0)
+  // Sort the solutions under limits and find the one that is closest to ik_seed_state
+  if (solution_found)
   {
-    double mindist = DBL_MAX;
-    int minindex = -1;
-    std::vector<double> sol;
-
-    for (size_t i = 0; i < solutions_obey_limits.size(); ++i)
+    bool swap_flag = true;
+    std::vector<double> temp_sol;
+    double temp_dist;
+    for (std::size_t i = 0; i < solutions_obey_limits.size(); ++i)
     {
-      sol = solutions_obey_limits[i];
-      double dist = harmonize(ik_seed_state, sol);
-      ROS_INFO_STREAM_NAMED("ikfast", "Dist " << i << " dist " << dist);
-
-      if (minindex == -1 || dist < mindist)
+      swap_flag = false;
+      for (std::size_t j = 0; j < solutions_obey_limits.size() - 1; ++j)
       {
-        minindex = i;
-        mindist = dist;
+        if (dists[j + 1] < dists[j])
+        {
+          temp_sol = solutions_obey_limits[j];
+          temp_dist = dists[j];
+          solutions_obey_limits[j] = solutions_obey_limits[j + 1];
+          solutions_obey_limits[j + 1] = temp_sol;
+          dists[j] = dists[j + 1];
+          dists[j + 1] = temp_dist;
+          swap_flag = true;
+        }
       }
```

Comment:
I think your sorting code would look cleaner if you were to use the [sort algorithm](http://www.cplusplus.com/reference/algorithm/sort/) with a predicate function that compares the distances to the seed.  This will do though.